operator-framework/src/test/java/io/javaoperatorsdk/operator/dependent/prevblocklist/DeploymentDependent.java

```diff
@@ -0,0 +1,95 @@
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.PodSpecBuilder;
+import io.fabric8.kubernetes.api.model.PodTemplateSpecBuilder;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.ResourceRequirementsBuilder;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.fabric8.kubernetes.api.model.apps.DeploymentSpecBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.GenericKubernetesResourceMatcher;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.SSABasedGenericKubernetesResourceMatcher;
+
+@KubernetesDependent
+public class DeploymentDependent
+    extends CRUDKubernetesDependentResource<Deployment, PrevAnnotationBlockCustomResource> {
+
+  public static final String RESOURCE_NAME = "test1";
+
+  public DeploymentDependent() {
+    super(Deployment.class);
+  }
+
+  @Override
+  protected Deployment desired(
+      PrevAnnotationBlockCustomResource primary,
+      Context<PrevAnnotationBlockCustomResource> context) {
+
+    return new DeploymentBuilder()
+        .withMetadata(
+            new ObjectMetaBuilder()
+                .withName(primary.getMetadata().getName())
+                .withNamespace(primary.getMetadata().getNamespace())
+                .build())
+        .withSpec(
+            new DeploymentSpecBuilder()
+                .withReplicas(1)
+                .withSelector(
+                    new LabelSelectorBuilder().withMatchLabels(Map.of("app", "nginx")).build())
+                .withTemplate(
+                    new PodTemplateSpecBuilder()
+                        .withMetadata(
+                            new ObjectMetaBuilder().withLabels(Map.of("app", "nginx")).build())
+                        .withSpec(
+                            new PodSpecBuilder()
+                                .withContainers(
+                                    new ContainerBuilder()
+                                        .withName("nginx")
+                                        .withImage("nginx:1.14.2")
+                                        .withResources(
+                                            new ResourceRequirementsBuilder()
+                                                .withLimits(Map.of("cpu", new Quantity("2000m")))
+                                                .build())
+                                        .build())
+                                .build())
+                        .build())
+                .build())
+        .build();
+  }
+
+  // for testing purposes replicating the matching logic but with the special matcher
```

Comment:
This should use the configuration option from #2760 instead.

Reply:
If that is merged, can change it (might be a new PR). thx!